### PR TITLE
Return different exit code if repo is locked or does not exist

### PR DESCRIPTION
**Pull request by MichaelEischer**
What does this PR change? What problem does it solve?
-----------------------------------------------------
Return exit code 11 if a repository cannot be locked and exit code 10 if the repository does not exist. The existence check relies on the presence of the `config` files. Other errors like an invalid password still result in exit code 1.

The new exit codes effectively form a category of exit codes (10+) that are related to the repository itself. The 1+ group contains the main exit codes.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Address two use cases from https://github.com/restic/restic/issues/956 .

Checklist
---------

- [x] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [x] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [x] I have added documentation for relevant changes (in the manual).
- [x] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [x] I have run `gofmt` on the code in all commits.
- [x] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [x] I'm done! This pull request is ready for review.

### CODE REVIEW

**Comment by MichaelEischer** [comment]
I've changed the exit codes to 10 and 11. With a bit of luck this might allow establishing at least a minimal kind of ordering in the exit codes.

File: doc/075_scripting.rst [hunk 1]
```diff
@@ -21,23 +21,51 @@ Check if a repository is already initialized
 ********************************************
 
 You may find a need to check if a repository is already initialized,
-perhaps to prevent your script from initializing a repository multiple
-times. The command ``cat config`` may be used for this purpose:
+perhaps to prevent your script from trying to initializing a repository multiple
```
Comment (rawtaz) [inline doc/075_scripting.rst:24]:
```suggestion
perhaps to prevent your script from trying to initialize a repository multiple
```

**Review by MichaelEischer — COMMENTED** [review]
LGTM.

**Review by rawtaz — DISMISSED** [review]

File: doc/075_scripting.rst (hunk 1 above)
Comment (MichaelEischer) [reply doc/075_scripting.rst]: fixed. thanks!

**Review by MichaelEischer — COMMENTED** [review]
LGTM.

**Pull request merged**